src/string_buffer.rs

```diff
@@ -80,22 +79,33 @@ impl StringBuffer {
     // Note that it is possible for this operation to take a long time in
     // pathological cases (lots of nodes, few line breaks).
     pub fn cur_offset(&self) -> usize {
-        unsafe {
-            let result = (&*self.last).cur_offset();
-            result.unwrap_or_else(|| {
-                println!("FIXME(#3) Unimplemented cur_offset across node boundaries");
-                0
-            })
-        }
+        // Try the last node first.
+        let result = unsafe {
```

Comment:
I would make the unsafe block enclose all the code in these two methods - an unsafe block should extend until the invariants are re-established, in this case that is until result goes out of scope.